src/cyhy_db/models/scan_doc.py

```diff
@@ -0,0 +1,120 @@
+    latest: bool = Field(default=True)
+    owner: str = Field(...)
+    snapshots: List[Link["SnapshotDoc"]] = Field(default=[])
+    source: str = Field(...)
+    time: datetime = Field(default_factory=utcnow)
+
+    @model_validator(mode="before")
+    def calculate_ip_int(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """Calculate the integer representation of an IP address."""
+        # ip may still be string if it was just set
+        values["ip_int"] = int(ip_address(values["ip"]))
+        return values
+
+    class Settings:
+        """Beanie settings to be used during testing."""
+
+        # These settings are intended for use only during testing.  See
+        # Abstract_Settings below.
+
+        name = "PyTest_ScanDocs"
+
+    class Abstract_Settings:
+        """Beanie settings to be inherited by subclasses."""
+
+        # This class is intentionally not named "Settings" to prevent Beanie from
+        # automatically applying these indices, which would result in the creation of
+        # an unnecessary collection. Instead, subclasses should define their own
+        # "Settings" class and include these indices along with any additional
+        # subclass-specific indices.
+
+        indexes = [
+            IndexModel(
+                [("latest", ASCENDING), ("ip_int", ASCENDING)], name="latest_ip"
+            ),
+            IndexModel([("time", ASCENDING), ("owner", ASCENDING)], name="time_owner"),
+            IndexModel([("int_ip", ASCENDING)], name="int_ip"),
+            IndexModel([("snapshots", ASCENDING)], name="snapshots", sparse=True),
+        ]
+
+    @classmethod
+    async def reset_latest_flag_by_owner(cls, owner: str):
+        """Reset the latest flag for all scans for a given owner."""
+        # flake8 E712 is "comparison to True should be 'if cond is True:' or 'if
+        # cond:'" but this is unavoidable due to Beanie syntax.
+        await cls.find(cls.latest == True, cls.owner == owner).update_many(  # noqa E712
+            Set({cls.latest: False})
+        )
+
+    @classmethod
+    async def reset_latest_flag_by_ip(
+        cls,
+        ips: (
+            int
+            | IPv4Address
+            | Iterable[int]
+            | Iterable[IPv4Address]
+            | Iterable[str]
+            | str
+        ),
+    ):
+        """Reset the latest flag for all scans for a given IP address."""
+        if isinstance(ips, Iterable):
+            ip_ints = [int(ip_address(x)) for x in ips]
+        else:
+            ip_ints = [int(ip_address(ips))]
+
+        # flake8 E712 is "comparison to True should be 'if cond is True:' or 'if
+        # cond:'" but this is unavoidable due to Beanie syntax.
+        await cls.find(
+            cls.latest == True, In(cls.ip_int, ip_ints)  # noqa E712
+        ).update_many(Set({cls.latest: False}))
+
+    @classmethod
+    async def tag_latest(
+        cls, owners: List[str], snapshot: Union["SnapshotDoc", ObjectId, str]
```

Comment:
Remove unneeded quotes:
```suggestion
        cls, owners: List[str], snapshot: Union[SnapshotDoc, ObjectId, str]
```